cpu/stm32f1/periph/gpio.c

```diff
@@ -207,6 +224,8 @@ void gpio_toggle(gpio_t pin)
 
 void gpio_write(gpio_t pin, int value)
 {
+    GPIO_INTERCEPT_WRITE(pin, value);
+
```

Comment:
Here and in all other calls to this macro. I don't get clearly what this macro does (I know what it does because I looked at the whole code, but this snip is not self explanatory at all), especially if the following code is being executed. I'd say it needs to be surrounded by `#ifdef MODULE_GPIO_EXP` (or the name that corresponds) and add a return to explicitly exit from this function. I see that this might be tedious since there are a lot of calls, but unless someone have a better idea I'd do it that way.

Reply:
I think the outline of the trade-offs in my response above also applies here.